### PR TITLE
txnbuild: Provide option to still generate V0 transaction envelopes

### PR DESCRIPTION
**Pull request by tamirms**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Provide option to still generate V0 transaction envelopes.

### Why

A developer may still want to generate v0 transactions. For example, if you need to generate a transaction and sign it with a hardware wallet. The hardware wallet may still be stuck on protocol 12. So, having the option to generate v0 transactions is convenient in this case.

### Known limitations

[N/A]

### CODE REVIEW

File: txnbuild/fee_bump_test.go [hunk 1]
```diff
@@ -50,15 +50,14 @@ func TestFeeBumpUpgradesV0Transaction(t *testing.T) {
 			BaseFee:              2 * MinBaseFee,
 			Memo:                 MemoText("test-memo"),
 			Timebounds:           NewInfiniteTimeout(),
+			V0:                   true,
```
Comment (leighmcculloch) [inline txnbuild/fee_bump_test.go:53]: Booleans in APIs are difficult to extend and if we have a V2 and so on we will have many. Could this be an enumeration?

```
type TransactionVersion int
const (
	TransactionVersionLatest TransactionVersion = iota
	TransactionVersion0
	TransactionVersion1
)
```

**Comment by tamirms** [comment]
@abuiles tested signing v1 transactions on ledger and trezor hardware wallets and it turns out that there are no issues with signing v1 transactions. It seems like we don't have anymore use cases for generating v0 transactions.

I will close the PR because allowing v0 transactions does complicate the API a bit and we should have some use cases before we implement that feature in txnbuild

**Comment by leighmcculloch** [comment]
What about someone who needs to decode a V0 transaction and sign it and re-encode it. Does punting on this prevent us from doing that or that is still possible?

**Comment by tamirms** [comment]
> What about someone who needs to decode a V0 transaction and sign it and re-encode it. Does punting on this prevent us from doing that or that is still possible?

That is still possible. It is true that`NewTransaction()` only constructs v1 transactions. However, you can still obtain a v0 transaction by calling `TransactionFromXDR()` to decode a v0 transaction envelope string.

Signing and encoding transactions will work for both v0 and v1 transaction types: 

https://github.com/stellar/go/blob/release-horizonclient-v3.2.0/txnbuild/transaction.go#L156-L174

**Pull request closed without merging**